### PR TITLE
augmenting protoschema with info for parsing

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Adding additional information to Protobuf schema so that the protobuf can be more easily parsed

### CODE REVIEW

File: pulsar-client-schema/src/test/java/org/apache/pulsar/client/schema/ProtobufSchemaTest.java [hunk 1]
```diff
@@ -67,4 +78,33 @@ public void testSchema() {
...
+                    Collections.emptyMap());
+        } catch (Exception e) {
+            Assert.fail();
+        }
+
+        try {
+            ProtobufSchema<org.apache.pulsar.client.schema.proto.Test.TestMessage> protobufSchema
+                    = ProtobufSchema.ofGenericClass(String.class,
+                    Collections.emptyMap());
+            Assert.fail();
```
Comment (sijie) [inline pulsar-client-schema/src/test/java/org/apache/pulsar/client/schema/ProtobufSchemaTest.java]: nit: put a meaningful message in fail method 

e.g.
`fail("Should not construct a ProtobufShema over a non-protobuf-generated class")`

File: pulsar-client-schema/src/test/java/org/apache/pulsar/client/schema/ProtobufSchemaTest.java [hunk 2]
```diff
@@ -67,4 +78,33 @@ public void testSchema() {
...
     }
+
+    @Test
+    public void testGenericOf() {
+        try {
+            ProtobufSchema<org.apache.pulsar.client.schema.proto.Test.TestMessage> protobufSchema
+                    = ProtobufSchema.ofGenericClass(org.apache.pulsar.client.schema.proto.Test.TestMessage.class,
+                    Collections.emptyMap());
+        } catch (Exception e) {
+            Assert.fail();
```
Comment (sijie) [inline pulsar-client-schema/src/test/java/org/apache/pulsar/client/schema/ProtobufSchemaTest.java]: nit: put a message in the fail method

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Pull request merged**